modules/core/04-channel/keeper/keeper.go

```diff
@@ -653,8 +665,42 @@ func (k Keeper) HasPruningSequenceStart(ctx sdk.Context, portID, channelID strin
 	return store.Has(host.PruningSequenceStartKey(portID, channelID))
 }
 
-// PruneAcknowledgements prunes packet acknowledgements from the store that have a sequence number less than or equal to the pruning sequence.
-// The number of packet acknowledgements pruned is equal to limit. Pruning only occurs after a channel has been upgraded.
-func (Keeper) PruneAcknowledgements(ctx sdk.Context, portID, channelID string, limit, pruningSequence uint64) error {
-	return nil
+// PruneAcknowledgements prunes packet acknowledgements and receipts that have a sequence number less than pruning sequence end.
+// The number of packet acks/receipts pruned is equal to limit. Pruning can only occur after a channel has been upgraded.
+//
+// Pruning sequence start keeps track of the packet ack/receipt that can be pruned next. When it reaches pruningSequenceEnd,
+// pruning is complete.
+func (k Keeper) PruneAcknowledgements(ctx sdk.Context, portID, channelID string, limit uint64) (uint64, uint64, error) {
+	if !k.HasPruningSequenceStart(ctx, portID, channelID) {
+		return 0, 0, errorsmod.Wrapf(types.ErrPruningSequenceStartNotFound, "port ID (%s) channel ID (%s)", portID, channelID)
+	}
+
+	pruningSequenceStart := k.GetPruningSequenceStart(ctx, portID, channelID)
```

Comment:
I find it odd that the get func here doesn't return a `found` bool. But I guess it is fine

Reply:
I agree, I found I had the same reaction too! 

But apparently the argument is saving gas with a `Has` check? :D lol

Reply:
Isn't this increasing gas? I suspect the majority of calls to this function to have the sequence set, which results in extra gas costs

Reply:
I think it would increase gas in the successful case of it being set. And save gas in the scenario of it not being set